feat(auth): добавляет файлы авторизации.

Nataliya2020 opened the pull request:
Создает компоненты авторизации Signin и Signup с валидацией, компонент Input с возможностью настраивать его с помощью props и компонент Button

Nataliya2020 committed:
feat(auth): добавляет файлы авторизации.

Создает компоненты авторизации Signin и Signup с валидацией, компонент Input с возможностью настраивать его с помощью props и компонент Button

WebHayk commented on src/pages/auth/regist/SignupPage.tsx:
```diff
@@ -0,0 +1,15 @@
+import Signup from '../../../components/auth/regist';
+import type {SubmitLoginProps} from "../../../core/interfaces/interfaces.ts";
+import signupPageCss from './SignupPage.module.css';
```
можно назвать просто styles, это же внутри signup, понятно что касается signup

WebHayk commented on src/App.tsx:
```diff
@@ -0,0 +1,32 @@
+function App() {
+  const [page, setPage] = useState<'login' | 'registr'>('registr');
+
+  const handleClickButton = (e: React.MouseEvent<HTMLButtonElement>): void => {
```
лучше использовать полные названия -> event, лучше читается код и поддерживается

WebHayk reviewed (commented):
regist -> register, лучше написать полные названия.

Nataliya2020 committed:
refactor: улучшает читаемость и поддержку кода

Переименовывает: - названия папок "regist" в полные "register", - в компоненте названия импорта стилей из модуля css в "styles",
- параметр события с "e" на "event".
